src/freenet/clients/http/FProxyToadlet.java

```diff
@@ -320,9 +320,11 @@ private static void addDownloadOptions(ToadletContext ctx, HTMLNode optionList,
 				        netLevel == NETWORK_THREAT_LEVEL.MAXIMUM))
 				filterChecked = false;
 		}
+		boolean downloadDisabled = !core.isDownloadDisabled()
+			&& (!ctx.getContainer().publicGatewayMode() || ctx.isAllowedFullAccess());
```

Comment:
Regarding: `boolean downloadDisabled = !core.isDownloadDisabled()`

Is it disabled or not disabled? This logic seems contradicting; or is the naming flawed?

Assuming the naming is OK, per the PR title, I would expect the logic to read:

```java
boolean downloadDisabled = 
        // download is either disabled fully on this node
        core.isDownloadDisabled()
        // or we're accessing in public gateway mode and do not have full access
        || (ctx.getContainer().publicGatewayMode() && !ctx.isAllowedFullAccess())
```
(this would match the previous authorization logic in this class, combined with the authorization logic in QueueToadlet)

Reply:
You’re right -- thank you for catching this!

Reply:
This is now fixed, and many more places check for public gateway mode. Thank you again!